### DISCUSSION TITLE
rtsp video

### DISCUSSION THREAD

**Discussion by Bconley123** [whitphx/streamlit-webrtc#570] [Q&A]
hello i'd like to run streamlit-webrtc on my pc and have on same network a raspberry pi with a rtsp stream from its camera. 
Ive looked at the code examples but am having a hard time figuring out how to implement rtsp stream url into the code. I see how in streamlit the camera is selectable but is only for the camera the pc has, but i want to use opencv with the rtsp camera. Is there a way to do this?

**Comment by whitphx** [comment]
[The uni-directional (server-to-client) video streaming example](https://github.com/whitphx/streamlit-webrtc/blob/17f91d1656cf3e5414b281e0ab5968de82e33cff/app.py#L454) includes [the RTSP source](https://github.com/whitphx/streamlit-webrtc/blob/17f91d1656cf3e5414b281e0ab5968de82e33cff/app.py#L477) which does not use OpenCV though.

If you want to use OpenCV anyway, there is another option like https://discuss.streamlit.io/t/possible-to-access-webcam-realtime-in-a-streamlit-app/947/5?u=whitphx while it uses `st.image` to show the frames which is less efficient to transmit data over the network.
